cmd/srcd/cmd/components_test.go

```diff
@@ -0,0 +1,161 @@
+
+	expected := regexp.MustCompile(`srcd/cli-daemon:\S+ +yes +yes +srcd-cli-daemon`)
+	s.Regexp(expected, out.String())
+}
+
+func (s *ComponentsTestSuite) TestInstall() {
+	require := s.Require()
+
+	out, err := s.RunCommand(context.TODO(), "components", "list")
+	require.NoError(err, out.String())
+
+	// Get the exact image:version of gitbase
+	exp := regexp.MustCompile(`(srcd/gitbase:\S+) +(yes|no)`)
+	matches := exp.FindStringSubmatch(out.String())
+
+	require.NotNil(matches)
+	require.Len(matches, 3)
+
+	imgVersion := matches[1]
+	installed := matches[2]
+
+	// If installed, remove it
+	if installed == "yes" {
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+		defer cancel()
+		err = docker.RemoveImage(ctx, imgVersion)
+		require.NoError(err)
+	}
+
+	// Check it's not installed
+	out, err = s.RunCommand(context.TODO(), "components", "list")
+	require.NoError(err, out.String())
+
+	expected := regexp.MustCompile(`srcd/gitbase:\S+ +no +no +srcd-cli-gitbase`)
+	require.Regexp(expected, out.String())
+
+	// Install
+	out, err = s.RunCommand(context.TODO(), "components", "install", "srcd/gitbase")
+	require.NoError(err, out.String())
+
+	// Check it's installed
+	out, err = s.RunCommand(context.TODO(), "components", "list")
+	require.NoError(err, out.String())
+
+	expected = regexp.MustCompile(`srcd/gitbase:\S+ +yes +no +srcd-cli-gitbase`)
+	require.Regexp(expected, out.String())
+
+	// Call install again, should be an exit 0
+	out, err = s.RunCommand(context.TODO(), "components", "install", "srcd/gitbase")
+	require.NoError(err, out.String())
+}
+
+func (s *ComponentsTestSuite) TestInstallAlias() {
+	require := s.Require()
+
+	// Install with image name
+	out, err := s.RunCommand(context.TODO(), "components", "install", "srcd/cli-daemon")
+	require.NoError(err, out.String())
+
+	// Install with container name
+	out, err = s.RunCommand(context.TODO(), "components", "install", "srcd-cli-daemon")
+	require.NoError(err, out.String())
+}
+
+func (s *ComponentsTestSuite) TestInstallUnknown() {
+	require := s.Require()
+
+	// Call install with a srcd image not managed by engine
+	out, err := s.RunCommand(context.TODO(), "components", "install", "srcd/lookout")
+	require.Error(err)
+	require.Contains(out.String(), "srcd/lookout is not valid. Component must be one of")
+}
+
+func (s *ComponentsTestSuite) TestInstallVersion() {
+	require := s.Require()
+
+	out, err := s.RunCommand(context.TODO(), "components", "list")
+	require.NoError(err, out.String())
+
+	// Get the exact image:version of gitbase
+	exp := regexp.MustCompile(`(srcd/gitbase:\S+)`)
+	matches := exp.FindStringSubmatch(out.String())
+
+	require.NotNil(matches)
+	require.Len(matches, 2)
+
+	imgVersion := matches[1]
+
+	// Call install with image:version
+	out, err = s.RunCommand(context.TODO(), "components", "install", imgVersion)
+	require.Error(err)
+	require.Contains(out.String(), imgVersion+" is not valid. Component must be one of")
```

Comment:
we're using many times similar regexp to test similar things (here and there) ;
what if we extract it and make it more readable?

considering https://play.golang.org/p/bC_LNM02Hi- it could be:
```go
out, err := s.RunCommand(context.TODO(), "components", "list")
parts, err := extract(out.String())
require.NoError(err)
imgVersion := parts[version]
require.Equals(parts[installed], "no")
...
out, err = s.RunCommand(context.TODO(), "components", "install", imgVersion)
parts, err = extract(out.String())
require.NoError(err)
require.Equals(parts[version], imgVersion)
```

Reply:
sounds good. Maybe a bit overkill for the current needs, but that would make it easier to build future tests. Feel free to do a PR with this.